File: src/main/java/br/uema/application/util/Response.java
package br.uema.application.util;

import org.springframework.http.HttpStatus;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;

import java.util.HashMap;
import java.util.Map;

public class Response {

    private Map<String, Object> meta;

    private Object data;

    public Response(HttpStatus status, String message, Object data){
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();

        Map<String, Object> map = new HashMap<String, Object>();
        map.put("code", status.value());
        map.put("message", message);
        map.put("authorization", authentication.getCredentials());

        this.meta = map;
        this.data = data;

    }
}

File: src/main/java/br/uema/application/util/Response.kt
package br.uema.application.util

import org.springframework.http.HttpStatus
import org.springframework.security.core.Authentication
import org.springframework.security.core.context.SecurityContextHolder

class Response() {

    var meta : Map<String, Any?>? = null
    var data : Any? = null

    constructor(status: HttpStatus, message: String?, data: Any?) : this() {

        val authentication: Authentication? = SecurityContextHolder.getContext().authentication

        var authorization = authentication?.credentials

        this.meta = hashMapOf("code" to status.value(), "message" to message, "authorization" to authorization)
        this.data = data
    }
}